pkg/expr/commands.go

```diff
@@ -171,6 +171,8 @@ func (gr *ReduceCommand) Execute(_ context.Context, vars mathexp.Vars) (mathexp.
 				Text:     fmt.Sprintf("Reduce operation is not needed. Input query or expression %s is already reduced data.", gr.VarToReduce),
 			})
 			newRes.Values = append(newRes.Values, copyV)
+		case mathexp.NoData:
+			newRes.Values = append(newRes.Values, v)
```

Comment:
The copying is so pointers don't accidentally mutate data in upstream operations (since the result of each operation can be returned in SSE).

Metadata is still a bit wonky and undefined. But in general, for the _output_ of a non-datasource query operation such as reduce, it is probably better to create a NewNoData() that creates a Frame with no fields.

Side Note: My idea with "NoData" is "untyped nodata". In the future, I could see adding NoDataTimeSeries etc.